app/database/operator/base_data_operator/index.ts

```diff
@@ -92,6 +92,10 @@ export default class BaseDataOperator {
 
                 // We found a record in the database that matches this element; hence, we'll proceed for an UPDATE operation
                 if (existingRecord) {
+                    if (shouldUpdate && !shouldUpdate(existingRecord, newElement)) {
+                        continue;
+                    }
+
```

Comment:
In useEffect of Permalinks, we are fetchingPostAround, where we are using handlepost operator where on processedPosts function we use these lines, which pushes empty array of posts.

Reply:
The problem here is that we are using the models returned from the `handlePosts`, which is no longer returning all the posts, but only the updated ones.

I think the easiest way to solve this is to get the posts from the database on `app/actions/remote/post.ts` `fetchPostsAround` after the batching. I would go with using the same we do for chunks (from some time to other time) instead of passing the list of ids, which is going to be more taxing for the database.

@Muhammadqasim001 Is this something you are working on? Will you be open to create a PR for this? Or should I take a look at it?

Reply:
This is a regression and I think is best if we fix it now and cherry pick for 2.8

FYI @amyblais

Reply:
Will open a PR tomorrow morning